timesketch/api/v1/resources/exportstream.py

```diff
@@ -0,0 +1,191 @@
+        )
+        return OpenSearchDataStore(pool_maxsize=pool_maxsize)
+
+    @login_required
+    def post(self, sketch_id):
+        """Handles POST request to export events.
+
+        Handler for /api/v1/sketches/:sketch_id/exportstream/
+
+        Note:
+            This endpoint is optimized for streaming large volumes of event
+            data and is not intended as a substitute for the interactive
+            `/explore` endpoint. While it supports basic filtering to narrow
+            down the export scope, complex search features are not available.
+
+        Args:
+            sketch_id (int): Integer primary key for a sketch database model.
+
+        Returns:
+            Streamed response of events in JSONL format.
+
+        Raises:
+            HTTPException:
+                - 400 (BAD_REQUEST): If the form data is invalid or if no
+                  valid search indices are found for the export.
+                - 403 (FORBIDDEN): If the user does not have read permissions
+                  for the sketch, or if the sketch is archived.
+                - 404 (NOT_FOUND): If the sketch with the given ID does not
+                  exist.
+        """
+        sketch = Sketch.get_with_acl(sketch_id)
+        if not sketch:
+            abort(HTTP_STATUS_CODE_NOT_FOUND, "No sketch found with this ID.")
+
+        if not sketch.has_permission(current_user, "read"):
+            abort(
+                HTTP_STATUS_CODE_FORBIDDEN,
+                "User does not have read access controls on sketch.",
+            )
+
+        if sketch.get_status.status == "archived":
+            abort(HTTP_STATUS_CODE_FORBIDDEN, "Unable to query on an archived sketch.")
+
+        form = forms.ExploreForm.build(request)
+        if not form.validate_on_submit():
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Unable to export data, unable to validate form data",
+            )
+
+        query_dsl = form.dsl.data
+        query_filter = request.json.get("filter") or {}
+        return_field_string = form.fields.data
+
+        # Resolve return fields
+        if return_field_string:
+            return_fields = [x.strip() for x in return_field_string.split(",")]
+        else:
+            return_fields = query_filter.get("fields", [])
+            return_fields = [field["field"] for field in return_fields]
+            if not return_fields:
+                return_fields = DEFAULT_SOURCE_FIELDS
+
+        # Resolve Indices
+        all_timeline_ids = [t.id for t in sketch.timelines]
+        indices = query_filter.get("indices", all_timeline_ids)
+        if "_all" in indices:
+            indices = all_timeline_ids
+
+        # Validate indices and get timeline IDs (checks permissions implicitly)
+        indices, timeline_ids = utils.get_validated_indices(indices, sketch)
+
+        # Map to actual OpenSearch index names for the PIT
+        indices_for_pit = []
```

Comment:
```suggestion
        timeline_ids_set = set(timeline_ids)
        indices_for_pit = [
            t.searchindex.index_name
            for t in sketch.timelines
            if t.id in timeline_ids_set
        ]
```

might be more efficiant? Convert timeline_ids to a set before iterating over all timelines to map them to OpenSearch indices. This changes the inner loop check from slow $O(N)$ (linear time) to fast $O(1)$ (constant time)